### PR TITLE
Add sample for Over the Air from Azure IoT

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Add Over the Air azure sample

## Motivation and Context

- Add Over the Air azure sample
- Full pattern including application to upload code in blob storage

## Types of changes

- [ ] Improvement (non-breaking change that improves a sample)
- [ ] Bug fix (fixes an issue with a current sample)
- [x] New Sample (adds a new sample)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Signed-off-by: Ellerbach <laurelle@microsoft.com>

### CODE REVIEW

File: samples/AzureSDK/AzureEdgeOta/AzureEdgeOTAEngine/Program.cs [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    // We won't do anything
+    // This global try catch is to make sure whatever happen, we will safely be able to go
+    // To sleep
+    Trace(ex.ToString());
+    GoToSleep();
+}
+
+Thread.Sleep(Timeout.InfiniteTimeSpan);
+// Just go to sleep when we will arrive at this point
+GoToSleep();
```
Comment (josesimoes) [inline samples/AzureSDK/AzureEdgeOta/AzureEdgeOTAEngine/Program.cs:117]: Execution will neve reach here...

File: samples/AzureSDK/AzureEdgeOta/AzureEdgeOTAEngine/Program.cs [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+}
+
+void DownloadBinaryFile(string url, string sas)
+{
+    int bytesRead = 0;
+    int totalBytes = 0;
+    HttpWebResponse httpWebResponse;
+    string fileName = url.Substring(url.LastIndexOf('/') + 1);
+
+    var httpWebRequest = (HttpWebRequest)WebRequest.Create($"{url}?{sas}");
```
Comment (josesimoes) [inline samples/AzureSDK/AzureEdgeOta/AzureEdgeOTAEngine/Program.cs:298]: You should use the new HttpClient instead, it's sooo much cleaner.

Comment (Ellerbach) [reply samples/AzureSDK/AzureEdgeOta/AzureEdgeOTAEngine/Program.cs:298]: right, I started this code before :-D So will adjust then

**Review by josesimoes — APPROVED** [review]
LGMT!

**Pull request merged**